packages/flutter_tools/lib/src/base/terminal.dart

```diff
@@ -30,9 +31,50 @@ enum TerminalColor {
   grey,
 }
 
+final OutputPreferences _kOutputPreferences = OutputPreferences();
+
+OutputPreferences get outputPreferences => (context == null || context[OutputPreferences] == null)
+    ? _kOutputPreferences
+    : context[OutputPreferences];
+
+/// A class that contains the context settings for command text output to the
+/// console.
+class OutputPreferences {
+  OutputPreferences({
+    bool wrapText,
+    int wrapColumn,
+    bool showColor,
+  })  : wrapText = wrapText ?? io.stdio?.hasTerminal ?? const io.Stdio().hasTerminal,
+        wrapColumn = wrapColumn ?? io.stdio?.terminalColumns ?? const io.Stdio().terminalColumns ?? kDefaultTerminalColumns,
+        showColor = showColor ?? platform.stdoutSupportsAnsi ?? false;
+
+  /// If [wrapText] is true, then output text sent to the context's [Logger]
+  /// instance (e.g. from the [printError] or [printStatus] functions) will be
+  /// wrapped to be no longer than the [wrapColumn] specifies. Defaults to true.
```

Comment:
Some of the docs in this page needs updating

Reply:
Which ones are you referring to?  I looked over those three comments and didn't see anything that wasn't correct. I updated the formatting of `promptForCharInput`, since it didn't match the prevailing style, but that was it.

Reply:
I elaborated on the details of the default values.